pkg/reconciler/vertex/scaling/client_cache.go

```diff
@@ -0,0 +1,65 @@
+package scaling
+
+import (
+	"container/list"
+	"sync"
+
+	daemonclient "github.com/numaproj/numaflow/pkg/daemon/client"
+)
+
+type entry struct {
+	address string
+	client  *daemonclient.DaemonClient
+}
+
+// ClientCache is a cache for daemon clients.
```

Comment:
please comment what kind of cache you have implemented. also.. can we make it generic?